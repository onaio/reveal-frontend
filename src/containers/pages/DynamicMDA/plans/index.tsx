import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPLansReducer, {
  genericFetchPlans,
  makeGenericPlansArraySelector,
  reducerName as GenericPLansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlan } from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(GenericPLansReducerName, GenericPLansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for Dynamic MDA plans */
const MDAPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a list of MDA plans */
const MDAPLansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPLansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_PLANS,
  pageUrl: REPORT_MDA_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
};

MDAPLansList.defaultProps = defaultProps;

export { MDAPLansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = MDAPlansArraySelector(state, {
    interventionTypes: [InterventionType.DynamicMDA],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected MDA Plans component */
const ConnectedMDAPLansList = connect(mapStateToProps, mapDispatchToProps)(MDAPLansList);

export default ConnectedMDAPLansList;
