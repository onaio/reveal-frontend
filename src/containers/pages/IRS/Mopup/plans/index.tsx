import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { MOP_UP_REPORTING_TITLE } from '../../../../../configs/lang';
import { IRS_MOP_UP_REPORT_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSMopUpReporting = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: MOP_UP_REPORTING_TITLE,
    pageUrl: IRS_MOP_UP_REPORT_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
