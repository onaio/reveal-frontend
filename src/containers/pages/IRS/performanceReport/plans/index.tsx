import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { IRS_PERFORMANCE_REPORTING_TITLE } from '../../../../../configs/lang';
import { PERFORMANCE_REPORT_IRS_PLAN_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSPerfomenceReport = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: IRS_PERFORMANCE_REPORTING_TITLE,
    pageUrl: PERFORMANCE_REPORT_IRS_PLAN_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
