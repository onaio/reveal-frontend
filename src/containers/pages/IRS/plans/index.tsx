import ListView from '@onaio/list-view';
import reducerRegistry from '@onaio/redux-reducer-registry';
import React, { useEffect, useState } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Col, Row } from 'reactstrap';
import { Store } from 'redux';
import HeaderBreadcrumb from '../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
import Loading from '../../../../components/page/Loading';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { HOME, HOME_URL, IRS_PLANS, REPORT_IRS_PLAN_URL } from '../../../../constants';
import supersetFetch from '../../../../services/superset';
import IRSPlansReducer, {
  fetchIRSPlans,
  getIRSPlansArray,
  IRSPlan,
  reducerName as IRSPlansReducerName,
} from '../../../../store/ducks/generic/plans';

/** register the plan definitions reducer */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);

/** interface for PlanList props */
interface PlanListProps {
  fetchPlans: typeof fetchIRSPlans;
  plans: IRSPlan[];
  service: typeof supersetFetch;
}

/** Simple component that loads the new plan form and allows you to create a new plan */
const IRSPlansList = (props: PlanListProps) => {
  const { fetchPlans, plans, service } = props;
  const [loading, setLoading] = useState<boolean>(true);

  const pageTitle: string = IRS_PLANS;

  const breadcrumbProps = {
    currentPage: {
      label: pageTitle,
      url: REPORT_IRS_PLAN_URL,
    },
    pages: [
      {
        label: HOME,
        url: HOME_URL,
      },
    ],
  };

  /** async function to load the data */
  async function loadData() {
    try {
      setLoading(plans.length < 1); // only set loading when there are no plans
      await service(SUPERSET_IRS_REPORTING_PLANS_SLICE).then((result: IRSPlan[]) =>
        fetchPlans(result)
      );
    } catch (e) {
      // do something with the error?
    } finally {
      setLoading(false);
    }
  }

  useEffect(() => {
    loadData();
  }, []);

  if (loading === true) {
    return <Loading />;
  }

  const listViewProps = {
    data: plans.map(planObj => {
      return [
        <Link to={`${REPORT_IRS_PLAN_URL}/${planObj.plan_id}`} key={planObj.plan_id}>
          {planObj.plan_title}
        </Link>,
        planObj.plan_date,
        planObj.plan_effective_period_start,
        planObj.plan_effective_period_end,
        planObj.plan_status,
      ];
    }),
    headerItems: ['Title', 'Date Created', 'Start Date', 'End Date', 'Status'],
    tableClass: 'table table-bordered plans-list',
  };

  return (
    <div>
      <Helmet>
        <title>{pageTitle}</title>
      </Helmet>
      <HeaderBreadcrumb {...breadcrumbProps} />
      <Row>
        <Col md={8}>
          <h3 className="mt-3 mb-3 page-title">{pageTitle}</h3>
        </Col>
      </Row>
      <Row>
        <Col>
          <ListView {...listViewProps} />
        </Col>
      </Row>
    </div>
  );
};

/** Declare default props for IRSPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchIRSPlans,
  plans: [],
  service: supersetFetch,
};

IRSPlansList.defaultProps = defaultProps;

export { IRSPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  plans: IRSPlan[];
}

/** map state to props */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const planDefinitionsArray = getIRSPlansArray(state);

  return {
    plans: planDefinitionsArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchIRSPlans };

/** Connected ActiveFI component */
const ConnectedIRSPlansList = connect(
  mapStateToProps,
  mapDispatchToProps
)(IRSPlansList);

export default ConnectedIRSPlansList;
