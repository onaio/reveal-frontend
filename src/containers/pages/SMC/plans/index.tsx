import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { HIDDEN_PLAN_STATUSES, SUPERSET_SMC_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { SMC_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_SMC_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams } from '../../../../helpers/utils';
import { RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import SMCPlansReducer, {
  fetchSMCPlans,
  makeSMCPlansArraySelector,
  reducerName as SMCPlansReducerName,
  SMCPLANType,
} from '../../../../store/ducks/generic/SMCPlans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the SMC plan definitions reducer */
reducerRegistry.register(SMCPlansReducerName, SMCPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);
// store.dispatch(fetchSMCPlans(testlans as any))

/** Simple component that loads a preview list of SMC plans */
const SMCPlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for SMCPlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: fetchSMCPlans,
  pageTitle: SMC_PLANS,
  pageUrl: REPORT_SMC_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_SMC_REPORTING_PLANS_SLICE,
};

SMCPlansList.defaultProps = defaultProps;

export { SMCPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: SMCPLANType[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const SMCPlansArray = makeSMCPlansArraySelector()(state, {
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: SMCPlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchSMCPlans };

/** Connected ActiveFI component */
const ConnectedSMCPlansList = connect(mapStateToProps, mapDispatchToProps)(SMCPlansList);

export default ConnectedSMCPlansList;
