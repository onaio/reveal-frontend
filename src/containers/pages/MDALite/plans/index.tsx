import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_LITE_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_LITE_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPlansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as genericReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(genericReducerName, GenericPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for MDA lite plans */
const makeMDALitePlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of MDA plans */
const MDALitePlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDALitePlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_LITE_PLANS,
  pageUrl: REPORT_MDA_LITE_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
};

MDALitePlansList.defaultProps = defaultProps;

export { MDALitePlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDALitePlansArray = makeMDALitePlansArraySelector(state, {
    interventionTypes: [InterventionType.MDALite],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: MDALitePlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedMDALitePlansList = connect(mapStateToProps, mapDispatchToProps)(MDALitePlansList);

export default ConnectedMDALitePlansList;
