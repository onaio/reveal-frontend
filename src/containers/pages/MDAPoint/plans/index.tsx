import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_POINT_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_POINT_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams } from '../../../../helpers/utils';
import { RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPlansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as genericReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(genericReducerName, GenericPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for MDA point plans */
const makeMDAPointPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of MDA plans */
const MDAPointPlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPointPlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_POINT_PLANS,
  pageUrl: REPORT_MDA_POINT_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE,
};

MDAPointPlansList.defaultProps = defaultProps;

export { MDAPointPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDAPointPlansArray = makeMDAPointPlansArraySelector(state, {
    interventionTypes: [InterventionType.MDAPoint],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: MDAPointPlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedMDAPointPlansList = connect(mapStateToProps, mapDispatchToProps)(MDAPointPlansList);

export default ConnectedMDAPointPlansList;
