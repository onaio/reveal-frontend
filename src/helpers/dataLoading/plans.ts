import { ActionCreator, AnyAction } from 'redux';
import { USER_HAS_NO_PLAN_ASSIGNMENTS } from '../../configs/lang';
import { OPENSRP_PLANS, OPENSRP_PLANS_BY_USER_FILTER } from '../../constants';
import { OpenSRPService } from '../../services/opensrp';
import store from '../../store';
import { AddPlanDefinitionAction } from '../../store/ducks/opensrp/PlanDefinition';
import { fetchPlansByUser, FetchPlansByUserAction } from '../../store/ducks/opensrp/planIdsByUser';
import { FetchPlanRecordsAction, PlanPayload, PlanRecordResponse } from '../../store/ducks/plans';
import { displayError } from '../errors';
import { extractPlanRecordResponseFromPlanPayload } from '../utils';

/** find plans that the given user has access to
 * @param {string} userName - username
 * @param {ActionCreator<FetchPlansByUserAction>} actionCreator - action creator for adding practitioners to store
 * @param {typeof OpenSRPService} service -  the OpenSRP service
 * @param {ActionCreator<AnyAction>} responseActionCreator - optionalActionCreator to dispatch the planData in response
 */
export async function loadPlansByUserFilter<T>(
  userName: string,
  actionCreator: ActionCreator<FetchPlansByUserAction> = fetchPlansByUser,
  service: typeof OpenSRPService = OpenSRPService,
  responseActionCreator?: ActionCreator<AnyAction>
) {
  const serve = new service(`${OPENSRP_PLANS_BY_USER_FILTER}/${userName}`);
  serve
    .list()
    .then((response: T[] | null) => {
      if (response === null) {
        return Promise.reject(new Error(USER_HAS_NO_PLAN_ASSIGNMENTS));
      }
      store.dispatch(actionCreator(response, userName));
      if (responseActionCreator) {
        store.dispatch(responseActionCreator(response));
      }
    })
    .catch((err: Error) => {
      displayError(err);
    });
}

/** fetch plans payload from the opensrp api
 * @param {OpenSRPService} service - openSRPService
 * @param {ActionCreator<FetchPlanRecordsAction>} actionCreator - action creator for fetchPlanRecords
 * @param {Dispatch<SetStateAction<boolean>>} - setState function
 */
export const loadOpenSRPPlans = (
  service: typeof OpenSRPService,
  actionCreator: ActionCreator<FetchPlanRecordsAction>,
  setLoading: React.Dispatch<React.SetStateAction<boolean>>
) => {
  const OpenSrpPlanService = new service(OPENSRP_PLANS);
  OpenSrpPlanService.list()
    .then((plans: PlanPayload[]) => {
      const extractedPlanRecords = plans
        .map(plan => extractPlanRecordResponseFromPlanPayload(plan))
        .filter(plan => !!plan);
      actionCreator(extractedPlanRecords as PlanRecordResponse[]);
      setLoading(false);
    })
    .catch(err => {
      setLoading(false);
      displayError(err);
    });
};

/** fetch single plan payload from the opensrp api
 * @param {string} planId - the plan's id
 * @param {OpenSRPService} service - openSRPService
 * @param {ActionCreator<AddPlanDefinitionAction>} actionCreator - action creator for addPlanDefinition
 */
export async function loadOpenSRPPlan(
  planId: string,
  service: typeof OpenSRPService,
  actionCreator: ActionCreator<AddPlanDefinitionAction>
) {
  const server = new service(OPENSRP_PLANS);
  return server
    .read(planId)
    .then(planFromAPI => {
      const currentPlan = Array.isArray(planFromAPI) ? planFromAPI[0] : planFromAPI;
      actionCreator(currentPlan);
    })
    .catch(err => {
      displayError(err);
    });
}
