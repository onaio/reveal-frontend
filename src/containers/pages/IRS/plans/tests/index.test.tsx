import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import { IRSPlansList } from '../';
import { IRSPlan } from '../../../../../store/ducks/generic/plans';
import * as fixtures from '../../../../../store/ducks/generic/tests/fixtures';

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

const history = createBrowserHistory();

describe('components/IRS Reports/IRSPlansList', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const props = {
      plans: fixtures.plans as IRSPlan[],
    };
    shallow(
      <Router history={history}>
        <IRSPlansList {...props} />
      </Router>
    );
  });

  it('renders plan definition list correctly', () => {
    fetch.mockResponseOnce(JSON.stringify({}));
    const props = {
      plans: fixtures.plans as IRSPlan[],
    };
    const wrapper = mount(
      <Router history={history}>
        <IRSPlansList {...props} />
      </Router>
    );
    expect(toJson(wrapper.find('BreadcrumbItem li'))).toMatchSnapshot('breadcrumbs');
    expect(toJson(wrapper.find('h3.page-title'))).toMatchSnapshot('page title');
    expect(toJson(wrapper.find('thead tr th'))).toMatchSnapshot('table headers');
    expect(toJson(wrapper.find('tbody tr td'))).toMatchSnapshot('table rows');
    wrapper.unmount();
  });
});
