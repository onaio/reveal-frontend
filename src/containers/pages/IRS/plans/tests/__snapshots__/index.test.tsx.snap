// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: breadcrumbs 1`] = `
Array [
  <li
    className="breadcrumb-item"
  >
    <Link
      key="0"
      replace={false}
      to="/"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </Link>
  </li>,
  <li
    aria-current="page"
    className="active breadcrumb-item"
  >
    IRS Plans
  </li>,
]
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: page title 1`] = `
<h3
  className="mt-3 mb-3 page-title"
>
  IRS Plans
</h3>
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table headers 1`] = `
Array [
  <th
    className="listview-td"
    key="0"
  >
    Title
  </th>,
  <th
    className="listview-td"
    key="1"
  >
    Date Created
  </th>,
  <th
    className="listview-td"
    key="2"
  >
    Start Date
  </th>,
  <th
    className="listview-td"
    key="3"
  >
    End Date
  </th>,
  <th
    className="listview-td"
    key="4"
  >
    Status
  </th>,
]
`;

exports[`components/IRS Reports/IRSPlansList renders plan definition list correctly: table rows 1`] = `
Array [
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="727c3d40-e118-564a-b231-aac633e6abce"
      replace={false}
      to="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce"
    >
      <a
        href="/intervention/irs/report/727c3d40-e118-564a-b231-aac633e6abce"
        onClick={[Function]}
      >
        IRS 2019-09-05 TEST
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    2019-09-05
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-09-05
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-25
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    retired
  </td>,
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="02e358ed-87fe-5720-8782-0afe28cb7e8c"
      replace={false}
      to="/intervention/irs/report/02e358ed-87fe-5720-8782-0afe28cb7e8c"
    >
      <a
        href="/intervention/irs/report/02e358ed-87fe-5720-8782-0afe28cb7e8c"
        onClick={[Function]}
      >
        IRS 2019-08-30m
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    2019-08-30
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-08-30
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-14
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    active
  </td>,
  <td
    className="listview-th"
    key="0"
  >
    <Link
      key="c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
      replace={false}
      to="/intervention/irs/report/c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
    >
      <a
        href="/intervention/irs/report/c6e59ec1-8c54-5fdf-acf3-ee3692983f38"
        onClick={[Function]}
      >
        Berg Namibia 2019
      </a>
    </Link>
  </td>,
  <td
    className="listview-th"
    key="1"
  >
    2019-08-29
  </td>,
  <td
    className="listview-th"
    key="2"
  >
    2019-08-29
  </td>,
  <td
    className="listview-th"
    key="3"
  >
    2019-09-18
  </td>,
  <td
    className="listview-th"
    key="4"
  >
    active
  </td>,
]
`;
